js/views/errorScreen.js

```diff
@@ -13,9 +13,20 @@ var ErrorScreen = React.createClass({
   render: function() {
     var errorTitle, errorDescription, errorAction;
     if (CONSTANTS.ERROR_MESSAGE.hasOwnProperty(this.props.errorCode.code)){
+
+      var errorAction = CONSTANTS.SKIN_TEXT.ERROR_ACTION;
+      if( CONSTANTS.ERROR_MESSAGE[this.props.errorCode.code].action ){
+        errorAction = CONSTANTS.ERROR_MESSAGE[this.props.errorCode.code].action;
+      }
+
       errorTitle = Utils.getLocalizedString(this.props.language, CONSTANTS.ERROR_MESSAGE[this.props.errorCode.code].title, this.props.localizableStrings)
       errorDescription = Utils.getLocalizedString(this.props.language, CONSTANTS.ERROR_MESSAGE[this.props.errorCode.code].description, this.props.localizableStrings);
-      errorAction = Utils.getLocalizedString(this.props.language, CONSTANTS.SKIN_TEXT.ERROR_ACTION, this.props.localizableStrings);
+      /// TODO - need to make countdown functionality display for all languages
+      if (this.props.errorCode.code === "future" && this.props.language === "en"){
+        errorDescription = "This video will be available in " + Utils.getStartCountdown(this.props.errorCode.flight_start_time * 1000 - Date.now());
```

Comment:
Can we validate this.props.errorCode.flight_start_time, in case we get some unexpected values

Reply:
@aeng7 added some checks